main.go

```diff
@@ -152,6 +163,8 @@ func runAPIPolling(done chan error, url, token string, yamlConfig YamlConfig, re
 	}
 
 	var jobs []queryJob
+	var configs []queryConfig
+	var jobsMutex sync.RWMutex
```

Comment:
nit: the jobsMutex and jobs fields can be extracted to a type that ensures concurrency safe operations.